megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectFireHandler.java

```diff
@@ -456,24 +448,30 @@ else if ((null != bestSpotter) && !(this instanceof ArtilleryWeaponDirectFireHan
 
         // artillery may unintentially clear minefields, but only if it wasn't
         // trying to
-        if (!mineClear && game.containsMinefield(targetPos)) {
-            Enumeration<Minefield> minefields = game.getMinefields(targetPos)
-                    .elements();
-            ArrayList<Minefield> mfRemoved = new ArrayList<Minefield>();
-            while (minefields.hasMoreElements()) {
-                Minefield mf = minefields.nextElement();
-                if (server.clearMinefield(mf, ae, 10, vPhaseReport)) {
-                    mfRemoved.add(mf);
-                }
-            }
-            // we have to do it this way to avoid a concurrent error problem
-            for (Minefield mf : mfRemoved) {
-                server.removeMinefield(mf);
-            }
+        if (!mineClear) {
+            clearMineFields(targetPos, Minefield.CLEAR_NUMBER_WEAPON_ACCIDENT, vPhaseReport);
         }
 
         return false;
     }
+    
+    /**
+     * Worker function that clears minefields.
+     */
+    private void clearMineFields(Coords targetPos, int targetNum, Vector<Report> vPhaseReport) {
```

Comment:
Consolidates mine clearing code into one place. I was going to have FAE code invoke it as well, but then realized that FAE probably can't clear minefields (the "can I clear minefields" code appears to imply that only standard munitions can clear mine fields)

Reply:
I checked with the rules team and FAE can clear minefields per the rules in TO. For all the hexes the FAE covers. So they are pretty good minesweepers.

Reply:
Ok, I'll make the necessary changes.